src/callbacks/check_gradients.cpp

```diff
@@ -64,40 +65,44 @@ DataType compute_objective_function(model& m) {
 
 } // namespace
 
-check_gradients
-  ::check_gradients(DataType step_size,
-                                   bool verbose,
-                                   bool error_on_failure)
-  : m_step_size(step_size),
+check_gradients::check_gradients(std::set<execution_mode> modes,
+                                 DataType step_size,
+                                 bool verbose,
+                                 bool error_on_failure)
+  : m_modes(std::move(modes)),
+    m_step_size(step_size),
     m_verbose(verbose),
     m_error_on_failure(error_on_failure) {}
 
-void check_gradients::on_test_end(model *m) {
+void check_gradients::do_check_gradients(model& m) const {
 
   // Get objects from model
-  lbann_comm *comm = m->get_comm();
-  auto mode = m->get_execution_mode();
-  const auto& layers = m->get_layers();
+  auto& comm = *m.get_comm();
+  const auto mode = m.get_execution_mode();
+  const auto& layers = m.get_layers();
+
+  // Return immediately if execution mode is invalid
+  if (!m_modes.empty() && m_modes.count(mode) == 0) { return; }
```

Comment:
"Invalid" is a bad word choice. This returns if the user specified particular execution modes _and_ the current mode was not specified.

Reply:
Reworded.